File: backend/services/difficulty_service.py
from sqlalchemy.orm import Session

from schemas.difficulty_schema import DifficultySchema
from models.difficulty import Difficulty


class DifficultyService:
    """ Difficulty service. """

    def __init__(self, db: Session):
        self.db = db

    def find_all_difficulties(self) -> list[DifficultySchema]:
        """ Find all difficulties. """
        return self.db.query(Difficulty).all()

    def find_difficulty_by_id(self, difficulty_id: int) -> DifficultySchema:
        """ Find a difficulty by id. """
        return self.db.query(Difficulty).filter(Difficulty.id == difficulty_id).first()

File: frontend/app/services/difficultyService.ts
import DifficultySchema, {DifficultyListSchema, DifficultyType} from "~/models/difficulty";
import {config} from "~/config";

const API_BASE_URL = 'http://localhost:8000';

/** Difficulty service. */
export class DifficultyService {

    /** Get all difficulties. */
    async getDifficulties(): Promise<DifficultyType[]> {
        const response = await fetch(`${API_BASE_URL}/difficulties`);
        return DifficultyListSchema.parse(await response.json());
    }

    /** Get a difficulty by ID. */
    async getDifficulty(difficultyId: number): Promise<DifficultyType> {
        const response = await fetch(`${API_BASE_URL}/difficulties/${difficultyId}`);
        return DifficultySchema.parse(await response.json());
    }
}
